### PR TITLE
Show warning when manifest is outdated

### PR DESCRIPTION
**Pull request by barryvdh**
This adds a check to the base view to show a warning when the manifest file in the public folder is different from the one in the vendor dir. This prevents broken components or missing updates.

image

The exception is also updated to make it more clear.
Related: #594

### CODE REVIEW

**Comment by taylorotwell** [comment]
Method name doesn't make sense to me. Make it something like "assetsAreCurrent" or "assetsOutOfDate"

**Comment by barryvdh** [comment]
Renamed to `assetsAreCurrent`

**Comment by tomswinkels** [comment]
Why not autoupdate the assets?

**Comment by barryvdh** [comment]
The browser usually runs with different permissions and there is no hook in composer to define in this package (only in the root package)

File: src/Telescope.php [hunk 1]
```diff
@@ -701,4 +702,21 @@ public static function ignoreMigrations()
...
+     *
+     * @throws \Illuminate\Contracts\Filesystem\FileNotFoundException
+     * @return bool
+     */
+    public static function assetsAreCurrent()
+    {
+        $publishedPath = public_path('vendor/telescope/mix-manifest.json');
+
+        if (! File::exists($publishedPath)) {
+            throw new \RuntimeException('The Telescope assets are not published. Please run: php artisan telescope:publish');
```
Comment (stancl) [inline src/Telescope.php]: Could this be a custom exception (like `TelescopeAssetsNotPublishedException`) that implements `ProvidesSolution`?

Comment (barryvdh) [reply src/Telescope.php]: Technically it could yeah

Comment (barryvdh) [reply src/Telescope.php]: Not sure what isn't clear about this error and the solution though :P

Comment (stancl) [reply src/Telescope.php]: There could be a button to run `Artisan::call('telescope:publish')`, similarly to the migrate button when a table is not found.

**Comment by tomswinkels** [comment]
@barryvdh can you make a PR for Laravel Horizon?

**Comment by barryvdh** [comment]
I don't think @driesvints was convinced yet.

**Comment by tomswinkels** [comment]
@barryvdh but i think it is better to do the same in Horizon and Telescope :) @driesvints

**Comment by driesvints** [comment]
Sorry, I'm still opposed to do anything about this in the UI. But feel free to make the pr to see if Taylor will accept it.

**Pull request merged**